File: services/cookbook/service/src/main/java/family/haschka/wolkenschloss/cookbook/recipe/Group.java
package family.haschka.wolkenschloss.cookbook.recipe;

public enum Group {

    VOLUME("Volumes", "", Unit.CUBIC),
    WEIGHT("Weights", "", Unit.GRAM),
    KITCHEN("Kitchen Terms", "", Unit.BUNCH),
    COMMON("Common Terms", "", Unit.CAN);

    final String description;
    final String name;
    final Unit[] units;

    Group(String name, String description, Unit...units) {
        this.name = name;
        this.description = description;
        this.units = units;
    }
}

File: services/cookbook/core/src/main/kotlin/recipe/Group.kt
package family.haschka.wolkenschloss.cookbook.recipe

enum class Group(val title: String, val description: String, vararg val units: Unit) {
    VOLUME(
        "Volumes",
        "",
        Unit.CUBIC,
        Unit.LITER,
        Unit.DECILITER,
        Unit.CENTILITER,
        Unit.MILLILITER,
        Unit.FLUID_ONCE,
        Unit.CUP
    ),
    WEIGHT(
        "Weights",
        "",
        Unit.GRAM,
        Unit.DECIGRAM,
        Unit.MILLIGRAM,
        Unit.DEKAGRAM,
        Unit.KILOGRAM,
        Unit.POUND
    ),
    KITCHEN(
        "Kitchen Terms",
        "",
        Unit.BUNCH,
        Unit.DROP,
        Unit.DASH,
        Unit.SHOT,
        Unit.TEASPOON,
        Unit.TABLESPOON,
        Unit.MESSERSPITZE,
        Unit.PINCH
    ),
    COMMON(
        "Common Terms",
        "",
        Unit.CAN,
        Unit.CARTON,
        Unit.PIECE,
        Unit.PACKAGE,
        Unit.SLICE,
        Unit.SHEET);
}
